tests/unit/test_id_cli.py

```diff
@@ -56,7 +57,9 @@ def abc_src_dir(tmp_path_factory):
 )
 async def test_parse_ids_workflows(ids_in, ids_out):
     """It should parse workflows & tasks."""
-    workflows, _ = await parse_ids_async(*ids_in, constraint='workflows')
+    with patch.object(Path, 'exists') as mock_exists:
```

Comment:
There are a lot of very similar context managers here - I wonder if you could turn all these into a fixture?